client/my-sites/site-settings/form-general.jsx

```diff
@@ -418,12 +418,15 @@ export class SiteSettingsFormGeneral extends Component {
 			siteIsAtomic,
 			translate,
 		} = this.props;
-		const blogPublic = parseInt( fields.blog_public, 10 );
 
+		const blogPublic = parseInt( fields.blog_public, 10 );
 		const wpcomComingSoon = 1 === parseInt( fields.wpcom_coming_soon, 10 );
 		const wpcomPublicComingSoon = 1 === parseInt( fields.wpcom_public_coming_soon, 10 );
-
+		// Private and unlaunched for v1 Coming soon === coming soon mode v1 is on
+		const isPrivateAndUnlaunched = -1 === blogPublic && this.props.isUnlaunchedSite;
 		const isNonAtomicJetpackSite = siteIsJetpack && ! siteIsAtomic;
+		const isAnyComingSoonEnabled =
```

Comment:
yea, it looks like `site.is_coming_soon` works for both coming soon v1 and coming soon v2 so I don't understand why we need these flags?

Reply:
That's true. We still need the `isPrivateAndUnlaunched` check to catch v1 sites that are still in private and unlaunched mode. The option `wpcom_coming_soon` is not set in this case.

So it could be `site.is_coming_soon || isPrivateAndUnlaunched`

Reply:
Hey @ramonjd I actually have an edge case where `blogPublic == 1 and this.props.isUnlaunchedSite == true.` 😅 
5756475-zd-woothemes

Is there any reason we are checking both of these conditions instead of checking only if this site has launched?

![image](https://user-images.githubusercontent.com/3422709/205036053-9d88d2db-6a05-4890-8598-77788f1489ae.png)

Reply:
Actually it seems this problem is far deeper than this. The Backend Atomic API saving the API is not returning the proper result 😬

Reply:
>  I actually have an edge case where blogPublic == 1 and this.props.isUnlaunchedSite == true
> Is there any reason we are checking both of these conditions instead of checking only if this site has launched?

Very interesting 😄 Thanks for the ping @jdc91 

From memory, the check that line – `-1 === blogPublic` was to identify coming soon v1 sites, which were then "private" by default, that is, `blogPublic === -1`. Using that check we'd assume coming soon mode was to be activated in the settings UI.

However, public and unlaunched is a curious combination, and, at least back then, an ostensibly incompatible one. 

The intent was that all new sites started off as "unlaunched" and "public (not indexable") `blogPublic === 0`. 

The only thing I can think of is that something happened during a switch from simple to atomic?

Sorry if that is not very helpful!

Reply:
Thanks for the reply Ramon! It is definitely a weird combination and there definitely seems to be a backend fault as I mentioned above. Just went down a deep rabbit hole during a support rotation 🕳️ . Hopefully there is a simpler fix like a re transfer to Atomic.